src/Eloquent/Assets/AssetRepository.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Rapidez\Statamic\Eloquent\Assets;
+
+use Illuminate\Support\Facades\Cache;
+use Statamic\Contracts\Assets\Asset as AssetContract;
+use Statamic\Support\Str;
+use Statamic\Eloquent\Assets\AssetRepository as StatamicAssetRepository;
+
+class AssetRepository extends StatamicAssetRepository
+{
+    public function findById($id): ?AssetContract
+    {
+        [$container, $path] = explode('::', $id);
+
+        $filename = Str::afterLast($path, '/');
+        $folder = Str::contains($path, '/') ? Str::beforeLast($path, '/') : '/';
+
+        $cacheKey = "eloquent-asset-{$id}";
+
+        return Cache::rememberForever($cacheKey, function () use ($container, $filename, $folder) {
```

Comment:
Do we want this remembered forever? are there any potential issues doing that, getting stale data?
Or do relevant changes update the id in this case?

If we see issues we could potentially use flexible or a cache with an expiry